main.go

```diff
@@ -97,43 +97,49 @@ func main() {
 }
 
 func federate(ctx context.Context, w http.ResponseWriter, r *http.Request, apiClient v1.API) {
-	// TODO: extend to support multiple match queries
-	// This will only act based on first match query
 	params := r.URL.Query()
-	matchQuery := params.Get("match[]")
+	matchQueries := params["match[]"]
 
+	// TODO: shuld the timeout increase with the query count?
+	// Should it be per query?
+	// Should we make it configurable?
 	nctx, ncancel := context.WithTimeout(r.Context(), 2*time.Minute)
+	defer ncancel()
 	if params.Del("match[]"); len(params) > 0 {
 		nctx = addValues(nctx, params)
 	}
-	start := time.Now()
-	val, _, err := apiClient.Query(nctx, matchQuery, time.Now()) // Ignoring warnings for now.
-	responseTime := time.Since(start).Seconds()
-	ncancel()
+	for _, matchQuery := range matchQueries {
+		start := time.Now()
+		val, _, err := apiClient.Query(nctx, matchQuery, start) // Ignoring warnings for now.
```

Comment:
please move this comment to the above line, and clarify why we are ignoring this. a few words would be great. thanks.

Reply:
I can move it but the comment was there before so I don't really know about the why...

Reply:
I've moved it.